### PR TITLE
Switch monospace font from Space Mono to JetBrains Mono

### LINKED ISSUE #528: Selection of font "Space Mono" for code display is not ideal
image

On the Processing.org website ([example](https://processing.org/examples/graphing2dequation.html)), I have to take exception on the use of Colophon's "[Space Mono](https://fonts.google.com/specimen/Space+Mono)" for code. Code doesn't parse like regular text. The absence of any hook on the minuscule "j" in the style used makes it read like an "i". "i" and "j" are the most common names for iterating-variables for nested loops, and [that's been the case](https://stackoverflow.com/questions/4137785/why-are-variables-i-and-j-used-for-counters) since your grandfathers' time ("i" stood for "integer" or maybe "iterate" and then you'd use i, j, k, etc.). I can't for the life of me imagine why the CSS selects OpenType [stylistic set "ss02"](https://sparanoid.com/lab/opentype-features/), which features that truncated "j". Perhaps even more troubling is that the comma and period are very difficult to distinguish at the default size (14px), and naturally that's a fatal problem for reading code. Please find a suitable solution. This issue costs me and my students time basically every day. Removing "ss02" only resolves the "j" problem, but you're still left with the comma/period conundrum (I'm not sure what the use of "ss04" and "ss05" have in store for the user, but it's also worth investigating).

So, you may want to choose a different font or let Colophon help you revise the font or find some other solution. If you want me to be involved in figuring out an appropriate solution, let me know. I've already sent basically the same report to [Design Systems International](https://designsystems.international/), since they [appear to have been involved at one point](https://processing.org/people/), but I don't know who makes font decisions now.

https://processing.org/examples/graphing2dequation.html

### PR DESCRIPTION
**Pull request by SableRaf**
Replaced Space Mono with JetBrains Mono for improved legibility of code examples, and code snippets. Updated CSS to use JetBrains Mono as the default monospace font for all text which previously used Space Mono.

fixes #528

### CODE REVIEW

**Comment by SableRaf** [comment]
cc @pzelchenko @TechnicJelle for review

**Comment by SableRaf** [comment]
> I'm looking right now, and it seems okay, except the `l`, `t`, and `m` look weird...

Oooh good catch. This is due to the [OpenType glyph variants](https://github.com/JetBrains/JetBrainsMono/blob/master/README.md#opentype-features) (stylistic sets) which can be [controlled with CSS](https://github.com/JetBrains/JetBrainsMono/wiki/OpenType-features). I'll make a fix shortly. Thanks for your eagle eye!

**Comment by SableRaf** [comment]
Found the issue! The `font-feature-settings` property was set globally for the whole `html` tag in `base.css`. Fixed by creating a `--font-mono-settings` variable and setting it to `normal !important`.

File: src/styles/variables.css
Comment (Stefterv) [inline src/styles/variables.css:1]: Important is not a sustainable way to override styles, please avoid! 

If the css is not applied please make sure you have a more specific selector, see documentation here: https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity

**Review by Stefterv — APPROVED** [review]

**Comment by TechnicJelle** [comment]
> Oooh good catch. This is due to the [OpenType glyph variants](https://github.com/JetBrains/JetBrainsMono/blob/master/README.md?rgh-link-date=2024-07-11T08%3A46%3A49Z#opentype-features) (stylistic sets) which can be [controlled with CSS](https://github.com/JetBrains/JetBrainsMono/wiki/OpenType-features). I'll make a fix shortly. Thanks for your eagle eye!

Ohh, thanks! I didn't know about this feature! (Or, more likely, I did at one point, and then forgot... :sweat_smile: )
Then it looks all good to me too!

**Review by TechnicJelle — APPROVED** [review]

**Pull request merged**